### PR TITLE
React embed: fix modal close on click outside 

### PR DESCRIPTION
**Pull request by harshsinghatz**
## What does this PR do?

Refer to this issue:
[Issue](https://github.com/calcom/cal.com/issues/5093)

Fixes # (issue)
- Modal closes on click outside

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Test it on the normal success page

### CODE REVIEW

**Comment by harshsinghatz** [comment]
I have a doubt about the testing part for the embed, how can i test embed?. I have tested it on the main success page
I have tested this on the codesandbox [here](https://codesandbox.io/s/embed-popup-on-click-of-an-existing-element-demo-react-forked-6zrdpv?file=/src/App.js)

**Comment by hariombalhara** [comment]
@harshsinghatz  Thank you so much for your contribution. You can follow the instructions here 
https://github.com/calcom/cal.com/blob/main/packages/embeds/embed-core/README.md

Also, please make sure that you test Mobile and Desktop with Dark and Light themes in both Embed and Non Embed scenarios

**Comment by harshsinghatz** [comment]
The snapshots test seems to be failing(on main branch as well).
image

Should i update the snapshots?

**Comment by hariombalhara** [comment]
Yeah, you should run them with  --update-snapshots locally. No need to commit the snapshots. Technically we just ignore the snapshots.

We don't verify snapshots in CI currently(due to some limitations).

**Comment by harshsinghatz** [comment]
@hariombalhara i have tested the non embedded one.
The docs(provided above) didn't help me for testing the embedding one. Am I missing something, can you guide me on that?

**Comment by harshsinghatz** [comment]
@hariombalhara ??

**Comment by hariombalhara** [comment]
@harshsinghatz Can you help me with what didn't work for you exactly and what worked?

**Comment by harshsinghatz** [comment]
Ok sure,

What didn't work:
1. Currently we have implemented snapshot testing for the embed modal, which needs to updated constantly.so i don't think it is necessary to even have those (unnecessary increase in space for the repo).
2. I tried by adding "embed?embed=&embedType=modal" query in the url to open up the embed version in the browser but i was still not able to test the click outside. 

So a guide on how to test the embed locally(to see the changes in the working env) just as the we are able to do in the sandbox. If you can guide me i can do the changes in the docs as well.

**Comment by hariombalhara** [comment]
> Currently we have implemented snapshot testing for the embed modal, which needs to updated constantly.so i don't think it is necessary to even have those (unnecessary increase in space for the repo).
Actually, snapshot testing works locally. The only problem is that snapshot testing requires snapshots to be committed first and they are dependent on the machine they were generated on(Mac(different flavours), Linux(different flavours, Windows(different flavours)). So, it requires us to do a setup where CI would first generate the snapshots(on demand) and automatically commit those to our repo, so that whenever tests run in CI now, the new snapshots can be compared with the old one. This setup isn't trivial and we haven't spent enough time to get it running.
So, as I said snapshot updates need not be committed for now. 
_Considering it creates confusion for new contributors, I am open to a PR where we comment out snapshot generation code(with a TODO of implementing it back) and deletes snapshot files as well_ 


>I tried by adding "embed?embed=&embedType=modal" query in the url to open up the embed version in the browser but i was still not able to test the click outside.
You shouldn't test it this way. You need to follow the steps here https://github.com/calcom/cal.com/blob/main/packages/embeds/embed-core/README.md#running-tests as I mentioned earlier. This should automatically open a manual testing page on port 3100. That page has various embeds setup already that you can just test.
See this loom for a demo of how it works. https://www.loom.com/share/c9410f3765e24c57b189cf91e6304319

**Comment by harshsinghatz** [comment]
> 

Thanks @hariombalhara , got the point. Earlier the port 3100 didn't seem to working, ig i was not running the main app on 3000
I have tested it now. Below SM the touch on the side won't close the modal but i don't this that would cause that much problem because the space on the sides is very less.

File: apps/web/pages/success.tsx [hunk 1]
```diff
@@ -285,7 +285,7 @@ export default function Success(props: SuccessProps) {
       <BookingPageTagManager eventType={eventType} />
       <CustomBranding lightVal={props.profile.brandColor} darkVal={props.profile.darkBrandColor} />
       <main className={classNames(shouldAlignCentrally ? "mx-auto" : "", isEmbed ? "" : "max-w-3xl")}>
-        <div className={classNames("overflow-y-auto", isEmbed ? "" : "z-50 ")}>
+        <div className={classNames("overflow-y-auto sm:mx-auto sm:w-fit", isEmbed ? "" : "z-50 ")}>
```
Comment (hariombalhara) [inline apps/web/pages/success.tsx:288]: @harshsinghatz You might not need to change this styling. The way closing logic works is. It looks for an element having main class or a main tag itself. If click happens outside it, it closes the modal.

So, we need to appropriately place the `main` class on the element that's just filling the content.

I think the div element with classes "inline-block transform overflow-hidden rounded-lg" is the nearest. Try adding main class to it and fix any issue that come up after that.

Comment (harshsinghatz) [reply apps/web/pages/success.tsx]: Yeah, just adding the main class worked!

**Review by hariombalhara — APPROVED** [review]

**Comment by hariombalhara** [comment]
Thank you so much @harshsinghatz. Much appreciated 🙏

**Comment by harshsinghatz** [comment]
> Thank you so much @harshsinghatz. Much appreciated 🙏

My pleasure and thanks for guiding ik the changes are not much but I got familiar with the embeds package!

**Pull request merged**